### PR TITLE
[fix][client] Broker address resolution wrong if connect through a multi-dns names proxy

### PR DESCRIPTION
**Pull request by nicoloboschi**
### Motivation

Regression of https://github.com/apache/pulsar/pull/19327 (It only affects **dev** branches since no release have been done containing that pull)
in the client, to understand if is connected through a proxy, it checks the logical address vs the physical one, if they're different it supposes it has to target that specific broker passing through the proxy.

In the above pull request, the comparison logic is changed and the logical address is compared to ONE of the resolved dns names. in case of multiple dns names of the proxy, it might happens that the first resolved dns name doesn't match with the logical proxy address and therefore it's considered to be a broker dns name.
This leads to the proxy to recursive send connections to itself.

These are some logs extracted: 
```
11:02:35.141 [pulsar-client-io-2-1] INFO  org.apache.pulsar.client.impl.PulsarChannelInitializer - initializeClientCnx DEBUG pulsar-proxy:6651, pulsar-proxy.pcert.svc.cluster.local:6651                                                                                                                                                                              
11:02:35.147 [pulsar-client-io-2-1] INFO  org.apache.pulsar.client.impl.ConnectionPool - [[id: 0xce7e9e49, L:/10.42.0.247:41194 - R:pulsar-proxy.pcert.svc.cluster.local/10.43.217.3:6651]] Connected to server                                                                                                                                                        11:02:35.172 [pulsar-client-io-2-1] INFO  org.apache.pulsar.client.impl.ClientCnx - [id: 0xce7e9e49, L:/10.42.0.247:41194 - R:pulsar-proxy.pcert.svc.cluster.local/10.43.217.3:6651] Connected through proxy to target broker at pulsar-proxy:6651
```

### Modifications

* Revert the comparison logic changes while keeping the improvements of https://github.com/apache/pulsar/pull/19327

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by michaeljmarshall — APPROVED** [review]
Great work @nicoloboschi! Thank you for fixing this!

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java [hunk 1]
```diff
@@ -213,17 +213,16 @@ CompletableFuture<Channel> initSocks5IfConfig(Channel ch) {
 
     CompletableFuture<Channel> initializeClientCnx(Channel ch,
                                                    InetSocketAddress logicalAddress,
-                                                   InetSocketAddress resolvedPhysicalAddress) {
+                                                   InetSocketAddress resolvedPhysicalAddress,
+                                                   InetSocketAddress unresolvedPhysicalAddress) {
```
Comment (michaeljmarshall) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java:217]: We were using this address for `cnx.setRemoteHostName(...);` before. In order to maintain consistency, I think we should consider doing that in this PR too.

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
Requesting changes since I think we need to get the hostname right for the `setRemoteHostName` method.

**Comment by nicoloboschi** [comment]
> Requesting changes since I think we need to get the hostname right for the `setRemoteHostName` method.

Good point! I've fixed it PTAL

**Review by michaeljmarshall — APPROVED** [review]
LGTM

**Pull request merged**